meson.build

```diff
@@ -86,7 +86,8 @@ if get_option('portable') or host_machine.system() == 'windows'
     lite_bindir = '/'
     lite_docdir = '/doc'
     lite_datadir = '/data'
-elif host_machine.system() == 'darwin'
+elif get_option('bundle')
```

Comment:
This looks wrong to me. If we are on macOS no branch is going to be activated to define the lite_datadir & co. Normally we should ensure that if we are on macOS but without a bundle the condition above is activated.

Reply:
It should be satisfied in the "posix" `else`, though IMO I would keep the check for the `host_machine`.
The only thing I see here is the missing `bundle` option for the CI [script][1], which would break without it, but I guess I could do it in a commit later.

[1]: https://github.com/lite-xl/lite-xl/blob/master/scripts/build.sh#L51-L55

Reply:
> It should be satisfied in the "posix" else, though IMO I would keep the check for the host_machine.

My original thinking was if you wanted to create a bundle on a different operating system, maybe for distribution purposes, then it should be possible. But I agree that it's probably better to keep the check, just to make sure nothing breaks.

> The only thing I see here is the missing bundle option for the CI script, which would break without it, but I guess I could do it in a commit later.

If I add the operating system check, would it make sense to enable the bundle option by default in the script? Then, it would have the same functionality as before and would only build a bundle on macOS.

Reply:
> My original thinking was if you wanted to create a bundle on a different operating system, maybe for distribution purposes, then it should be possible. But I agree that it's probably better to keep the check, just to make sure nothing breaks.

I believe each bundle will require different prefixes/destination directories (see `scripts/appimage.shj` for example, so this should be a `macos-bundle` dedicated option.

> If I add the operating system check, would it make sense to enable the bundle option by default in the script? Then, it would have the same functionality as before and would only build a bundle on macOS.

No, if we want to provide a macOS bundle option that is disabled by default (which I don't agree if bundles are the default way to distribute mac applications nowadays) you should pass it also on the build script that uses it and I don't want to use posix builds after spent days on writing/testing scripts to produce a DMG image. So following your logic I would add a `disable-macos-bundle` to enable posix style builds instead, so no changes would be needed.

Reply:
> if we want to provide a macOS bundle option that is disabled by default (which I don't agree if bundles are the default way to distribute mac applications nowadays) 

That's a very good point @redtide. For the average macOS user, it would make more sense to generate a bundle by default, and provide them with an option to have the standard unix layout if they so choose.

This would also mean no changes have to be made to the CI scripts, and would make things a lot easier concerning this PR. Thoughts?

Reply:
Ok I've set the macOS bundle option to true, but it should only build the bundle on macOS. If you don't want the bundle, it's as simple as `-Dbundle=false` :)

Reply:
> This would also mean no changes have to be made to the CI scripts, and would make things a lot easier concerning this PR. Thoughts?

Yes, that is what I meant, meanwhile I enabled the CI for this PR, it had to run from the beginning to show the problem.

> Ok I've set the macOS bundle option to true, but it should only build the bundle on macOS. If you don't want the bundle, it's as simple as `-Dbundle=false` :)

This way it doesn't force any change being optional and should work as expected.